### PR TITLE
Add `legacyRemotingSecurityEnabled` flag

### LINKED ISSUE #616: Add a flag to explicitly disable `remotingSecurity`
### Is your feature request related to a problem? Please describe

At the moment, `remotingSecurity.enabled = true` is injected into CasC if the `controller.tag` is older than `2.326`.

This works when using the public Jenkins image, however it does not when building a custom image which does not follow the same versioning as the app itself, e.g. I'm building my own container which is at `v2.2.2` right now, even though I have a Jenkins `2.344` inside.

### Describe the solution you'd like

Add an explicit flag in the `values.yaml`, e.g. `legacyRemotingSecurityEnabled`. 

- Default: `null` (no behavior change, i.e. rely on `.Values.controller.tag` / `.Chart.AppVersion`)
- When `false`, do not include `remotingSecurity.enabled = true` regardless of the container version
- When `true`, include `remotingSecurity.enabled = true` regardless of the container version

### Describe alternatives you've considered

- I did consider just bumping my own container version to `v3`, but that's a hack.
- I tried overridding that in another CasC file, but the CasC plugin complains that the value is already defined. Maybe there's a case for doing the yaml overrides at the chart level? Or maybe that's already a feature that I'm not aware of?

### Additional context

Happy to PR.

### PR DESCRIPTION
**Pull request by dominykas**
### What this PR does / why we need it

- Implementation / reasoning described in #616
- ~~Also updates whitespace in the unit tests - they were failing for me locally. Do they even get executed in CI? I don't see anything in the logs (even though the command is mentioned in `ct.yaml`)?~~ It seems they're only failing for me locally... Is it due to helm version mismatch or smth? Seems CI is running 3.4, while I'm running 3.8...

### Which issue this PR fixes

- Fixes #616
- Alternative for #623 / Workaround for #592

### Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/unittests/jcasc-config-test.yaml [hunk 1]
```diff
@@ -1544,5 +1544,207 @@ tests:
                 usageStatisticsEnabled: true
             unclassified:
               location:
-                adminAddress: 
+                adminAddress:
                 url: http://RELEASE-NAME-jenkins:8080
+
+  - it: legacyRemotingSecurityEnabled = false (even though jenkins is 2.325 or older)
```
Comment (timja) [inline charts/jenkins/unittests/jcasc-config-test.yaml:1550]: what happens if the tag is not a semver tag and `legacyRemotingSecurityEnabled` is not set?

I can't easily tell from reading the code.

(would be good if there was a test showing what it does)

Comment (dominykas) [reply charts/jenkins/unittests/jcasc-config-test.yaml:1550]: Yeah, seems Helm does not have a nice way of handling this error or providing a nicer message... but adding that regex thing from #623 is also a bit overkill?

Comment (timja) [reply charts/jenkins/unittests/jcasc-config-test.yaml:1550]: can we just back out the semver thing.

Default this legacy flag to false, document that people on x version or older will need to enable this flag if they want to keep it enabled

Comment (dominykas) [reply charts/jenkins/unittests/jcasc-config-test.yaml:1550]: Sure, I don't mind doing that, but that means a breaking change and a major version bump?

**Review by timja — APPROVED** [review]

File: charts/jenkins/CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,10 @@ Use the following links to reference issues, PRs, and commits prior to v2.6.0.
 The change log until v1.5.7 was auto-generated based on git commits.
 Those entries include a reference to the git commit to be able to get more details.
 
+## 4.0.0
+
+Removes automatic `remotingSecurity` setting when using a container tag older than `2.236` (introduced in [`3.11.7`](#3117)). If you're using a version older than `2.236`, you should explicitly set `.controller.legacyRemotingSecurityEnabled` to `true`.
```
Comment (timja) [inline charts/jenkins/CHANGELOG.md:17]:
```suggestion
Removes automatic `remotingSecurity` setting when using a container tag older than `2.326` (introduced in [`3.11.7`](#3117)). If you're using a version older than `2.326`, you should explicitly set `.controller.legacyRemotingSecurityEnabled` to `true`. 
```

File: charts/jenkins/unittests/jcasc-config-test.yaml (hunk 1 above)
Comment (dominykas) [reply charts/jenkins/unittests/jcasc-config-test.yaml]: Cool, I'll open a separate PR once this one is merged, thanks

**Pull request merged**